src/contexts/pagination-context.ts

```diff
@@ -1,20 +1,21 @@
-import { PaginationContext, PaginationParams } from '../types'
+import { PaginationState } from '../types'
 
 /**
- * @param  {AlgoliaMultipleQueriesQuery} searchRequest
- * @param  {Context} options
+ * Create the current state of the pagination
+ *
+ * @param  {boolean} [finite]
+ * @param  {number} [hitsPerPage]
+ * @param  {number} [page]
  * @returns {SearchContext}
  */
-export function createPaginationContext({
-  paginationTotalHits,
-  hitsPerPage,
-  page,
-}: PaginationParams): // searchContext: SearchContext
-PaginationContext {
+export function createPaginationState(
+  finite?: boolean,
+  hitsPerPage?: number,
+  page?: number
+): PaginationState {
   return {
-    paginationTotalHits:
-      paginationTotalHits != null ? paginationTotalHits : 200,
     hitsPerPage: hitsPerPage === undefined ? 20 : hitsPerPage, // 20 is the Meilisearch's default limit value. `hitsPerPage` can be changed with `InsantSearch.configure`.
     page: page || 0, // default page is 0 if none is provided
+    finite: !!finite,
```

Comment:
Why do you use `!!` here?

Reply:
it forces the conversation to a boolean. So if it is a `falsy`, it becomes `false`

Example: 

`undefined`
! `undefined` => true
!! `undefined` => false

same goes for every falsy https://developer.mozilla.org/en-US/docs/Glossary/Falsy